### ISSUE TITLE
std-instead-of-core: there is no `core::collections::*`

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#14982]
Using the following flags

`--force-warn clippy::std-instead-of-core`

this code:

```rust
#![crate_type="lib"]
use std::{collections::HashMap, hash::Hash};

struct C;

trait Ctx {
    type BindGroupId;
}

impl Ctx for C {
    type BindGroupId = u32;
}

pub struct BindGroup {
    _id: <C as Ctx>::BindGroupId,
}

pub fn insert(map: &mut HashMap<*const BindGroup, u32>, bind_group: *const BindGroup) {
    map.insert(bind_group, 1);
}

```

caused the following diagnostics:

```
    Checking _85447 v0.1.0 (/tmp/icemaker_global_tempdir.t8dYaH8PqaqW/icemaker_clippyfix_tempdir.fAsrbYWloqmA/_85447)
warning: used import from `std` instead of `core`
 --> src/lib.rs:2:5
  |
2 | use std::{collections::HashMap, hash::Hash};
  |     ^^^ help: consider importing the item from `core`: `core`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#std_instead_of_core
  = note: requested on the command line with `--force-warn clippy::std-instead-of-core`

warning: `_85447` (lib) generated 1 warning (run `cargo clippy --fix --lib -p _85447` to apply 1 suggestion)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.11s

```

However after applying these diagnostics, the resulting code:

```rust
#![crate_type="lib"]
use core::{collections::HashMap, hash::Hash};

struct C;

trait Ctx {
    type BindGroupId;
}

impl Ctx for C {
    type BindGroupId = u32;
}

pub struct BindGroup {
    _id: <C as Ctx>::BindGroupId,
}

pub fn insert(map: &mut HashMap<*const BindGroup, u32>, bind_group: *const BindGroup) {
    map.insert(bind_group, 1);
}

```
no longer compiled:

```
    Checking _85447 v0.1.0 (/tmp/icemaker_global_tempdir.t8dYaH8PqaqW/icemaker_clippyfix_tempdir.fAsrbYWloqmA/_85447)
error[E0432]: unresolved import `core::collections`
 --> src/lib.rs:2:12
  |
2 | use core::{collections::HashMap, hash::Hash};
  |            ^^^^^^^^^^^ could not find `collections` in `core`

For more information about this error, try `rustc --explain E0432`.
error: could not compile `_85447` (lib test) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
error: could not compile `_85447` (lib) due to 1 previous error

```

Version:
```
rustc 1.89.0-nightly (076ec59ff 2025-06-05)
binary: rustc
commit-hash: 076ec59ff1dcf538b9d3a0b8e0d7f4edd0559959
commit-date: 2025-06-05
host: x86_64-unknown-linux-gnu
release: 1.89.0-nightly
LLVM version: 20.1.5

```
